local-cluster/tests/local_cluster.rs

```diff
@@ -2504,6 +2507,70 @@ fn run_test_load_program_accounts_partition(scan_commitment: CommitmentConfig) {
     );
 }
 
+#[test]
+#[serial]
+fn test_rpc_block_subscribe() {
+    let total_stake = 100 * DEFAULT_NODE_STAKE;
+    let leader_stake = total_stake;
+    let node_stakes = vec![leader_stake];
+    let mut validator_config = ValidatorConfig::default_for_test();
+    validator_config.enable_default_rpc_block_subscribe();
+
+    let validator_keys = vec![
+        "28bN3xyvrP4E8LwEgtLjhnkb7cY4amQb6DrYAbAYjgRV4GAGgkVM2K7wnxnAS7WDneuavza7x21MiafLu1HkwQt4",
+    ]
+    .iter()
+    .map(|s| (Arc::new(Keypair::from_base58_string(s)), true))
+    .take(node_stakes.len())
+    .collect::<Vec<_>>();
+
+    let mut config = ClusterConfig {
+        cluster_lamports: total_stake,
+        node_stakes,
+        validator_configs: vec![validator_config],
+        validator_keys: Some(validator_keys),
+        skip_warmup_slots: true,
+        ..ClusterConfig::default()
+    };
+    let cluster = LocalCluster::new(&mut config, SocketAddrSpace::Unspecified);
+    let (mut block_subscribe_client, receiver) = PubsubClient::block_subscribe(
+        &format!("ws://{}", &cluster.entry_point_info.rpc_pubsub.to_string()),
+        RpcBlockSubscribeFilter::All,
+        Some(RpcBlockSubscribeConfig {
+            commitment: Some(CommitmentConfig::confirmed()),
+            encoding: None,
+            transaction_details: None,
+            show_rewards: None,
+            max_supported_transaction_version: None,
+        }),
+    )
+    .unwrap();
+
+    let mut received_block = false;
+    loop {
```

Comment:
why not `while !received_block`
also should probably have a max timeout we fail on

Reply:
fixed! Yeah good idea.